### PR TITLE
Implemented new feature for enable audio , created options sound for All or only errors

### PR DESCRIPTION
**Pull request by SaurabPoudel**
- But there is a  warning : The AudioContext was not allowed to start. It must be resumed (or created) after a user gesture on the page.  I couldn't fix that

### CODE REVIEW

File: packages/keybr-textinput/lib/settings.ts [hunk 1]
```diff
@@ -59,10 +60,15 @@ export enum WhitespaceStyle {
...
+  All = 1,
+  Error = 2,
+}
 
 export const textDisplaySettings: TextDisplaySettings = {
   caretShapeStyle: CaretShapeStyle.Underline,
   caretMovementStyle: CaretMovementStyle.Smooth,
+  enableSoundStyle: EnableSoundStyle.All,
   whitespaceStyle: WhitespaceStyle.Bullet,
   sounds: false,
```
Comment (aradzie) [inline packages/keybr-textinput/lib/settings.ts:73]: Let's remove these two properties and merge them into one.
So,  remove the `enableSourndsStyle` and replace `sounds` with

```
  playSounds: PlaySounds.None,
```

File: packages/keybr-textinput/lib/settings.ts [hunk 2]
```diff
@@ -84,12 +90,18 @@ export const textDisplayProps = {
     WhitespaceStyle.Bullet,
   ),
   sounds: booleanProp("textDisplay.sounds", false),
+  enableSoundStyle: enumProp(
+    "textDisplay.enableSoundStyle",
+    EnableSoundStyle,
+    EnableSoundStyle.All,
+  ),
```
Comment (aradzie) [inline packages/keybr-textinput/lib/settings.ts:97]: Make these properties match the type definitions above.

`playSounds: enumProp("textDisplay.playSounds", PlaySounds, PlaySounds.None),`

File: packages/keybr-textinput/lib/settings.ts [hunk 3]
```diff
@@ -38,6 +38,7 @@ export function toTextInputSettings(settings: Settings): TextInputSettings {
 export type TextDisplaySettings = {
   readonly caretShapeStyle: CaretShapeStyle;
   readonly caretMovementStyle: CaretMovementStyle;
+  readonly enableSoundStyle: EnableSoundStyle;
   readonly whitespaceStyle: WhitespaceStyle;
   readonly sounds: boolean;
```
Comment (aradzie) [inline packages/keybr-textinput/lib/settings.ts:43]: readonly playSounds: PlaySounds;

File: packages/keybr-textinput/lib/settings.ts [hunk 4]
```diff
@@ -84,12 +90,18 @@ export const textDisplayProps = {
...
+    EnableSoundStyle,
+    EnableSoundStyle.All,
+  ),
 } as const;
 
 export function toTextDisplaySettings(settings: Settings): TextDisplaySettings {
   return {
     caretShapeStyle: settings.get(textDisplayProps.caretShapeStyle),
     caretMovementStyle: settings.get(textDisplayProps.caretMovementStyle),
+    enableSoundStyle: settings.get(textDisplayProps.enableSoundStyle),
```
Comment (aradzie) [inline packages/keybr-textinput/lib/settings.ts:104]: Please keep the order of properties match the order in the type definition.

File: packages/page-practice/lib/practice/Controller.tsx [hunk 5]
```diff
@@ -49,7 +50,7 @@ export const Controller = memo(function Controller({
 function usePracticeState(state: PracticeState) {
   const keyboard = useKeyboard();
   const [lines, setLines] = useState(state.lines); // Forces ui update.
-
+  const { settings } = useSettings();
```
Comment (aradzie) [inline packages/page-practice/lib/practice/Controller.tsx:53]: A big problem here. You use the `settings` variable` inside a hook, but this variable is not marked as a hook dependency. You should catch this error if you run linter `npm run lint`.

File: packages/page-practice/lib/practice/Controller.tsx [hunk 6]
```diff
@@ -90,16 +94,23 @@ function usePracticeState(state: PracticeState) {
...
-      playSound(TextInputSound.Click);
-      break;
-    case Feedback.Recovered:
-      playSound(TextInputSound.Click);
-      break;
-    case Feedback.Failed:
-      playSound(TextInputSound.Blip);
-      break;
+function playFeedbackSound(feedback: Feedback, option: number): void {
+  if (option === 1) {
```
Comment (aradzie) [inline packages/page-practice/lib/practice/Controller.tsx:98]: Please don't use magic numbers. Use `PlaySounds.None`, `PlaySounds.All`, etc.

**Review by aradzie — CHANGES_REQUESTED** [review]
Thanks for your PR!
I found a few issues in it. Let's fix those in the first round of review.

**Comment by aradzie** [comment]
Hey, I've made my own change-list to implement this feature -- https://github.com/aradzie/keybr.com/tree/sounds
Take inspiration from it

**Comment by aradzie** [comment]
Hey Saurab.

I know you wanted to contribute, but this is a simple change and it takes equal time to review your PR and to implement this feature. So I did it myself in [this commit](https://github.com/aradzie/keybr.com/commit/b699c22101a9dcb4e9db258ec3385121de2da033) 

Again, I am sorry I did not gave you the opportunity to contribute.

**Comment by SaurabPoudel** [comment]
@aradzie  haha cool

**Pull request closed without merging**